client/py/synnax/simulation/auto.py

```diff
@@ -0,0 +1,153 @@
+import time
+from typing import Protocol
+import synnax as sy
+
+
+class SimulatedResponse(Protocol):
+    def __init__(self):
+        ...
+
+    def step(self, state: dict[str, float]) -> dict[str, float]:
+        ...
+
+
+class Anomaly(Protocol):
+    def __init__(self):
+        ...
+
+    def step(self, state: dict[str, float]) -> dict[str, float]:
+        ...
+
+
+# All nominal state machines
+
+
+class TankPressure:
+    def __init__(
+        self, pt_channel: str, press_valve_channel: str, prevalve_channel: str
+    ) -> SimulatedResponse:
+        self.step_num = 0
+        self.pt = pt_channel
+        self.press = press_valve_channel
+        self.pre = prevalve_channel
+
+        # state[pt] = 0  # presssure
+        # state[press] = 0  # pressvalve, 0 closed, 1 open
+        # state[pre] = 1  # prevalve, 0 open, 1 closed
+
+    def step(self, state: dict[str, float]) -> dict[str, float]:
+        self.step_num += 1
+
+        if state[self.pt] >= 400:
+            state[self.press] = 0
+        else:
+            state[self.press] = 1
+
+        if state[self.press] == 1 and state[self.pre] == 1:
+            state[self.pt] += 1
+        elif state[self.press] == 0 and state[self.pre] == 0:
+            state[self.pt] -= 1
+        else:
+            state[self.pt] += 0
+
+        return state
+
+
+class Nominal:
+    def __init__(self, state_machine, error_step: int):
+        self.state_machine = state_machine
+        self.error_step = error_step
+
+    def step(self, state: dict[str, float], step_num: int):
+        self.state_machine.step(state)
+        return state
+
+
+# Anomaly Implementations
+
+
+class VoltageZero:
+    def __init__(self, state_machine, error_step: int):
+        self.state_machine = state_machine
+        self.error_step = error_step
+
+    def step(self, state: dict[str, float], step_num: int):
+        if step_num < self.error_step:
+            state[self.state_machine.pt] = -200
+        self.state_machine.step_num += 1
+        return state
+
+
+class VoltageFull:
+    def __init__(self, state_machine, error_step: int):
+        self.state_machine = state_machine
+        self.error_step = error_step
+
+    def step(self, state: dict[str, float], step_num: int):
+        if step_num < self.error_step:
+            state[self.state_machine.pt] = 1000
+        self.state_machine.step_num += 1
+        return state
+
+
+class PressValveStuckOpen:
```

Comment:
Both `PressValveStuckOpen` and `PreValveStuckOpen` are doing the same thing just with different valves. Can't we just take in the name of the valve we want to mess with and use a single class?